NAS2D/StringUtils.h

```diff
@@ -25,6 +26,8 @@ namespace NAS2D
 	{
 		if constexpr (std::is_same_v<T, std::string> || std::is_same_v<T, const char*>) {
 			return value;
+		} else if constexpr (std::is_convertible<T, std::string>::value) {
+			return value;
```

Comment:
As a bit of an aside, I think the `std::is_convertible` case would also cover the `const char*` case above. It might even cover the `std::string` case too, now that I think about it. We could potentially simplify. And the existing unit tests should let us know right away if it would work.

```cpp
if constexpr (std::is_convertible<T, std::string>::value) {
	return value;
} ...
```